### PR TITLE
Pip 1537 nested scatter

### PR DESCRIPTION
**Pull request by leepc12**
Hot fix for nested scatters.
There is no circleci testing for croo yet.
Tested with atac, chip, hic and some toy WDLs with subworkflows and nested scatters.

### CODE REVIEW

File: croo/cromwell_metadata.py [hunk 1]
```diff
@@ -137,31 +140,81 @@ def __parse_input_json(self):
...
                     )
                     continue
 
-                task_name = parent_wf_name + wf_name + '.' + task_alias
-                shard_idx = parent_wf_shard_idx + (shard_idx,)
+                none_free_parent_workflows = parent_workflows + (
+                    subworkflow_or_task_alias,
+                )
+                none_free_parent_workflows = (
+                    wf for wf in none_free_parent_workflows if wf
```
Comment (paul-sud) [inline croo/cromwell_metadata.py]: Should be explicit about not being `None`:
```suggestion
                    wf for wf in none_free_parent_workflows if wf is not None
```

File: croo/cromwell_metadata.py [hunk 2]
```diff
@@ -137,31 +140,81 @@ def __parse_input_json(self):
...
+                )
```
Comment (paul-sud) [inline croo/cromwell_metadata.py]: You could probably combine these into a single comprehension, saves reassigning.
```suggestion
                none_free_parent_workflows = (
                    workflow
                    for workflow
                    in parent_workflows + (subworkflow_or_task_alias,)
                    if workflow is not None
                )
```

**Review by paul-sud — CHANGES_REQUESTED** [review]
Some small comments. I think this PR would greatly benefit from tests too. You mentioned "a test toy problem with complex subworkflows", could you add it here? I don't think we need CircleCI right now but at the minimum adding tests with pytest would be good.

File: croo/cromwell_metadata.py (hunk 1 above)
Comment (leepc12) [reply croo/cromwell_metadata.py]: Fixed.

File: croo/cromwell_metadata.py (hunk 2 above)
Comment (leepc12) [reply croo/cromwell_metadata.py]: Fixed as you suggested.

Comment (leepc12) [reply croo/cromwell_metadata.py]: Yeah I think it's good time to add testing to Croo.
I will add basic linting and several integration tests to Croo.

**Comment by leepc12** [comment]
@paul-sud: I added pytests to test subworkflows and nested scatters. I will make a new ticket/PR for CircleCI and more refactoring.

**Review by paul-sud — COMMENTED** [review]
Overall looks OK, but we might want to consider storing the metadata JSONs for the tests instead of creating them every time. Then tests will run much faster and test fixtures will be simpler. You could still keep the WDL around in case you need to recreate them.

**Comment by leepc12** [comment]
@paul-sud Yep, fixed as you suggested.

**Review by paul-sud — APPROVED** [review]

**Pull request merged**